apps/voxelize/voxelize.cpp

```diff
@@ -38,31 +38,48 @@
 namespace
 {
 
+void _getNameAndExtension( const std::string& filePath,
+                           std::string& outputName, std::string& extension )
+{
+    outputName = filePath;
+    extension = ".mhd";
+
+    // Consider only the filename to check whether there is an extension or not
+    const std::string& fileName =
+            filePath.substr( filePath.find_last_of("/") + 1 );
```

Comment:
This is wrong, if there is no / find_last_of returns std::npos and I don't know what std::npos + 1 returns (if it's defined at all), but most probably 0.